server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -790,6 +791,18 @@ private String validateConfigurationValue(final String name, String value, final
             return null;
         }
 
+        if (type.equals(Integer.class) && NetworkModel.MACIdentifier.key().equalsIgnoreCase(name)) {
+            try {
+                final int val = Integer.parseInt(value);
+                if(val <0 || val >255){
+                    throw new InvalidParameterValueException(name+" value should be between 0 and 255. 0 value will disable this feature");
+                }
```

Comment:
Why are we using a magic value in the parameter to determine whether or not a feature is enabled?  Why not add an explicit flag to make it clearer for the end user?